command/cat.go

```diff
@@ -38,7 +39,9 @@ var catCommand = &cli.Command{
 		}
 		return err
 	},
-	Action: func(c *cli.Context) error {
+	Action: func(c *cli.Context) (err error) {
+		defer stat.Collect(c.Command.FullName(), &err)()
+
 		src, err := url.New(c.Args().Get(0))
```

Comment:
not sure if this is going to work. the `err` returned from `url.New` is actually the named return, or is it a new variable? if it's a new one, what will `stat.Collect` receive?

Reply:
for named returns it seems binding occurs before the return, so for `_, e := ur.New(); return e` value of `e` will be assigned to `err` before the return, and since `stat.Collect` has pointer to `err` it will receive error `e`

Reply:
👍